osu.Game/Input/Bindings/DatabasedKeyBindingContainer.cs

```diff
@@ -72,12 +88,11 @@ protected override void ReloadMappings()
                 KeyBindings = defaults;
             else
             {
-                KeyBindings = store.Query(ruleset?.ID, variant)
-                                   .OrderBy(b => defaults.FindIndex(d => (int)d.Action == b.IntAction))
-                                   // this ordering is important to ensure that we read entries from the database in the order
-                                   // enforced by DefaultKeyBindings. allow for song select to handle actions that may otherwise
-                                   // have been eaten by the music controller due to query order.
-                                   .ToList();
+                KeyBindings = realmKeyBindings.Detach()
+                                              // this ordering is important to ensure that we read entries from the database in the order
+                                              // enforced by DefaultKeyBindings. allow for song select to handle actions that may otherwise
+                                              // have been eaten by the music controller due to query order.
+                                              .OrderBy(b => defaults.FindIndex(d => (int)d.Action == b.ActionInt)).ToList();
```

Comment:
I haven't confirmed yet, but this may also need a `rulesetID` check.

Reply:
Never mind, the correct checks are in place above (https://github.com/ppy/osu/pull/11461/files#diff-cf554a9e1758b9acf3269373772131cdbdb3859005358eb30225af99f55d7217R57).